File: Expenses.Core.Entities/Purchase.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Expenses.Core.Entities
{
    public class Purchase
    {
        public Purchase ()
        {
            ProductList = new List<ProductPurchase>();
        }

        [Key]
        public int Id { get; set; }

        [Required]
        public Store Store { get; set; }
        public int StoreId { get; set; }

        [Required]
        public DateTime Date { get; set; }

        public ICollection<ProductPurchase> ProductList { get; set; }

        public double Total { get; set; }
    }
}

File: Expenses.UI/ClientApp/src/app/models/purchase.ts
import { ProductPurchase } from "./productPurchase";
import { Store } from "./store";

export class Purchase {
    idPurchase?: number;
    date: Date;
    dateString: string;
    store: Store;
    count: number;
    total: number;
    productList : ProductPurchase[]
  }